ipa-core/src/ff/curve_points.rs

```diff
@@ -35,16 +35,25 @@ impl SharedValue for RP25519 {
     const ZERO: Self = Self(CompressedRistretto([0_u8; 32]));
 }
 
+#[derive(thiserror::Error, Debug)]
+#[error("{0:?} is not the canonical encoding of a Ristretto point.")]
+pub struct NonCanonicalEncoding(CompressedRistretto);
+
 impl Serializable for RP25519 {
     type Size = <<RP25519 as SharedValue>::Storage as Block>::Size;
+    type DeserializationError = NonCanonicalEncoding;
 
     fn serialize(&self, buf: &mut GenericArray<u8, Self::Size>) {
         *buf.as_mut() = self.0.to_bytes();
     }
 
-    fn deserialize(buf: &GenericArray<u8, Self::Size>) -> Self {
-        debug_assert!(CompressedRistretto((*buf).into()).decompress().is_some());
-        RP25519(CompressedRistretto((*buf).into()))
+    fn deserialize(buf: &GenericArray<u8, Self::Size>) -> Result<Self, Self::DeserializationError> {
+        let point = CompressedRistretto((*buf).into());
+        if cfg!(debug_assertions) && point.decompress().is_none() {
+            return Err(NonCanonicalEncoding(point));
```

Comment:
I dont think the check will add that much since we are not sending curve points that often. However I think since the curve points are generated in the MPC protocol (using share conversion) and are not inputs by user devices or the report collector a debug assertion is fine, too.

Reply:
This will likely be easy to change later as we return an error now. `decompress` implementation is quite involved and the penalty to materialize may be significant. I would prefer to do it later if we want